packages/test/test-gc-sweep-tests/src/dataObjectWithChildDataObject.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { DataObjectFactory } from "@fluidframework/aqueduct";
+import { assert } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { DataObjectWithCounter, dataObjectWithCounterFactory } from "./dataObjectWithCounter";
+import { allFactories } from "./testDataObjects";
+
+const childKey = "childKey";
+export class DataObjectWithChildDataObject extends DataObjectWithCounter {
+    private unrefTimeStampMs?: number;
+    private oldChild?: DataObjectWithCounter;
+    private child?: DataObjectWithCounter;
+
+    public static get type(): string {
+        return "DataObjectWithChildDataObject";
+    }
+
+    protected async initializingFirstTime(props?: any): Promise<void> {
+        await super.initializingFirstTime(props);
+        await this.referenceNewChild();
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        await super.hasInitialized();
+        const childHandle = this.root.get<IFluidHandle<DataObjectWithCounter>>(childKey);
+        this.child = await childHandle?.get();
+    }
+
+    public stop() {
+        this.child?.stop();
+        this.isRunning = false;
+    }
+
+    public async unreferenceChild() {
+        assert(this.child !== undefined, `Child should be defined when unreferencing!`);
+        this.oldChild = this.child;
+        this.child.stop();
+        this.root.delete(childKey);
+        this.unrefTimeStampMs = Date.now();
+    }
+
+    public async referenceNewChild() {
+        this.child = await dataObjectWithCounterFactory.createInstance(this.context.containerRuntime);
+        await this.referenceChild(this.child.handle);
+    }
+
+    public async referenceChild(handle: IFluidHandle<DataObjectWithCounter>) {
+        this.oldChild = this.child;
```

Comment:
Why is oldChild set here? If an oldChild already exists, this will overwrite it.

Reply:
We are overwriting the oldChild so that we only maintain one reference to an old child. The undo redo stack is only going to be so big. Ours just happens to be the size of 1.